tendrl/alerting/api/manager.py

```diff
@@ -0,0 +1,91 @@
+from flask import Flask
+from flask import request
+import logging
+from multiprocessing import Event
+from multiprocessing import Process
+from tendrl.alerting.etcd.manager import EtcdInterface
+from tendrl.alerting.exceptions import AlertingError
+from tendrl.alerting.exceptions import InvalidRequest
+from tendrl.alerting.notification.exceptions import NotificationPluginError
+from tendrl.alerting.notification.exceptions import InvalidHandlerConfig
+from tendrl.alerting.etcd.exceptions import EtcdError
+from tendrl.alerting.notification.manager import NotificationPluginManager
+from tendrl.common.config import TendrlConfig
+
+
+config = TendrlConfig()
+LOG = logging.getLogger(__name__)
+app = Flask(__name__)
+
+
+@app.route("/alerting/notification_medium/supported")
+def get_handlers():
+    try:
+        return NotificationPluginManager().get_handlers()
+    except (
+        AlertingError,
+        NotificationPluginError
+    ) as ex:
+        raise ex
+
+
+@app.route("/alerting/notification_medium/<name>/help")
+def get_config_help(name):
+    try:
+        return NotificationPluginManager().get_config_help(name)
+    except (
+        InvalidRequest,
+        NotificationPluginError
+    ) as ex:
+        raise ex
+
+
+@app.route("/alerting/alerts")
+def get_alerts(filter_criteria=None):
+    try:
+        if len(request.args) > 0:
+            return str(EtcdInterface().get_alerts(request.args.items()))
+        return str(EtcdInterface().get_alerts())
+    except EtcdError as ex:
+        raise ex
+
+
+@app.route("/alerting/notification_medium/<name>/config", methods=['POST'])
+def add_destination(name):
+    try:
+        return NotificationPluginManager().add_destination(name, request.data)
+    except (
+        InvalidHandlerConfig,
+        NotificationPluginError
+    ) as ex:
+        raise ex
+
+
+@app.route("/alerting/alert_types")
+def get_alert_types():
+    try:
+        return str(EtcdInterface().get_alert_types())
+    except EtcdError as ex:
+        raise ex
+
+
+class APIManager(Process):
+
+    def __init__(self):
+        super(APIManager, self).__init__()
+        self._complete = Event()
+        self.host = config.get(
+            "alerting",
+            "server_interface"
+        )
+
+    def run(self):
+        try:
+            app.run(host=self.host, port=5001)
```

Comment:
Is the port configurable? If so externalize this as well in configuration. Default could be 5001 if not defined in configurations file.